genie-web/src/main/java/com/netflix/genie/web/services/impl/AgentRoutingServiceImpl.java

```diff
@@ -87,5 +104,63 @@ public void handleClientConnected(@NotBlank final String jobId) {
     public void handleClientDisconnected(@NotBlank final String jobId) {
         log.info("Agent executing job {} disconnected", jobId);
         this.agentConnectionPersistenceService.removeAgentConnection(jobId, genieHostInfo.getHostname());
+
+        for (AgentConnectionObserver agentConnectionObserver : agentConnectionObservers) {
+            this.connectionObserversNotificationExecutor.execute(
+                new AgentDisconnectedNotificationTask(agentConnectionObserver, jobId)
+            );
+        }
+    }
+
+    /**
+     * Task to deliver agent connected notification to a AgentConnectionObserver.
+     */
+    static class AgentConnectedNotificationTask implements Runnable {
```

Comment:
Use a lambda or method reference?

These 2 classes are 50 LOC (and object being created and destroyed at runtime) for two simple method calls...